### PR TITLE
Fix book form && add controllers for (authors & category)

### PR DESCRIPTION
**Pull request by fadyyousry**

### CODE REVIEW

File: app/controllers/books_controller.rb [hunk 1]
```diff
@@ -17,6 +17,9 @@ def edit
 
   def create
     @book = Book.new(book_params)
+    create_publisher
+    create_authors
+    create_categories
 
     respond_to do |format|
```
Comment (Magdyx) [inline app/controllers/books_controller.rb:24]: did you know that you can remove respond_to and format.html and the code will still works as intended?
remove it or keep it. Your call

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    def index
+      @categories = Category.all
+    end
+
+    def show
+      @books = @category.books
+    end
+
+    def destroy
+      @category.destroy
```
Comment (Magdyx) [inline app/controllers/categories_controller.rb:13]: do you know if destroy returns a boolean or nil?

File: app/controllers/books_controller.rb [hunk 3]
```diff
@@ -54,4 +63,49 @@ def book_params
...
+        author = Author.find_or_initialize_by(values)
+        if author.save
+          @book.authors.append(author)
+        end
+      end
+    end
+
+    def create_categories
+      categories_list.each do |values|
+        category = Category.find_or_initialize_by(values)
```
Comment (Magdyx) [inline app/controllers/books_controller.rb:105]: as per our conversation earlier, I prefer them in the model

File: app/controllers/categories_controller.rb (hunk 2 above)
Comment (fadyyousry) [reply app/controllers/categories_controller.rb:13]: It returns the deleted activerecord
image

File: app/controllers/books_controller.rb (hunk 3 above)
Comment (fadyyousry) [reply app/controllers/books_controller.rb:105]: ok

**Pull request merged**